packages/option/test/lion-option.test.js

```diff
@@ -0,0 +1,152 @@
+import { expect, fixture, html } from '@open-wc/testing';
+import sinon from 'sinon';
+
+import '../lion-option.js';
+
+describe('lion-option', () => {
+  describe('Values', () => {
+    it('has a modelValue', async () => {
+      const el = await fixture(html`
+        <lion-option .choiceValue=${10}></lion-option>
+      `);
+      expect(el.modelValue).to.deep.equal({ value: 10, checked: false });
+    });
+
+    it('can be checked', async () => {
+      const el = await fixture(html`
+        <lion-option .choiceValue=${10} checked></lion-option>
+      `);
+      expect(el.modelValue).to.deep.equal({ value: 10, checked: true });
+    });
+  });
+
+  describe('Accessibility', () => {
+    it('has the "option" role', async () => {
+      const el = await fixture(html`
+        <lion-option></lion-option>
+      `);
+      expect(el.getAttribute('role')).to.equal('option');
+    });
+
+    it('has "aria-selected" attribute when checked', async () => {
+      const el = await fixture(html`
+        <lion-option .choiceValue=${10} checked>Item 1</lion-option>
+      `);
+      expect(el.getAttribute('aria-selected')).to.equal('true');
+
+      el.checked = false;
+      // check that dom update is async
+      expect(el.getAttribute('aria-selected')).to.equal('true');
+
+      await el.updateComplete;
+      expect(el.getAttribute('aria-selected')).to.equal('false');
+    });
+
+    it('has "aria-disabled" and "disabled" attribute when disabled', async () => {
+      const el = await fixture(html`
+        <lion-option .choiceValue=${10} disabled>Item 1</lion-option>
+      `);
+      expect(el.getAttribute('aria-disabled')).to.equal('true');
+      expect(el.hasAttribute('disabled')).to.be.true;
+
+      el.disabled = false;
+      // check that dom update is async
+      expect(el.getAttribute('aria-disabled')).to.equal('true');
+      expect(el.hasAttribute('disabled')).to.be.true;
+
+      await el.updateComplete;
+      expect(el.getAttribute('aria-disabled')).to.equal('false');
+      expect(el.hasAttribute('disabled')).to.be.false;
+    });
+  });
+
+  describe('State reflection', () => {
+    it('can be active', async () => {
+      const el = await fixture(html`
+        <lion-option .choiceValue=${10}></lion-option>
+      `);
+      expect(el.active).to.equal(false);
+      expect(el.hasAttribute('active')).to.be.false;
+
+      el.active = true;
+      expect(el.active).to.be.true;
+      expect(el.hasAttribute('active')).to.be.false;
```

Comment:
separate test for `// check that dom update is async`

Reply:
@jorenbroekema renamed it to `adds asynchronous the attribute "active" when becomes active` as splitting the tests seemed to not help 🙈

Reply:
Maybe rename to "asynchronously adds the 'active' attribute when the element is active" for grammar :P

Reply:
changed